elyra/pipeline/processor_kfp.py

```diff
@@ -67,8 +97,22 @@ def process(self, pipeline):
             self.log.info("Kubeflow Pipeline successfully compiled.")
             self.log.debug("Kubeflow Pipeline was created in %s", pipeline_path)
 
-            # Upload the compiled pipeline and create an experiment and run
-            client = kfp.Client(host=api_endpoint)
+            # Upload the compiled pipeline, create an experiment and run
+
+            if (api_endpoint_username is not None) and (api_endpoint_password is not None):
```

Comment:
Is a password required for authentication?  Just wondering.

Reply:
@kevin-bates I'm also wondering if an empty string is a valid pwd. According to [dex kfdef yaml](https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_istio_dex.v1.0.2.yaml), this is the section we're looking at:
```
- kustomizeConfig:
      overlays:
      - istio
      parameters:
      - name: namespace
        value: auth
      - name: issuer
        value: http://dex.auth.svc.cluster.local:5556/dex
      - name: client_id
        value: kubeflow-oidc-authservice
      - name: oidc_redirect_uris
        value: '["/login/oidc"]'
      - name: static_email
        value: admin@kubeflow.org
      - name: static_password_hash
        value: $2y$12$ruoM7FqXrpVgaol44eRZW.4HWS8SAvg6KYVVSCIwKQPBmTpCm.EeO
      repoRef:
        name: manifests
        path: dex-auth/dex-crds
    name: dex
```
email and password can be edited, our environment is using the default kubeflow deployment.
We can try out with an empty pwd (only not sure what kind of hash it's being used but I can look it up)

Reply:
I'm not sure what I'm looking at here.  It looks like data values so I don't know how you'd determine if a password requires a value.  
> but I can look it up

Is there an API doc for the `authservice_session` thing?

Reply:
I wasn't able to find any specific documentation around that. 
I can try to add another auth user to our kfp env, where the pwd field is a bcrypt hash of an empty string. If that works, then in our code we would only check for `api_endpoint_username as you suggested. Otherwise if we also assume `api_endpoint_password` where empty string is actually its valid value, we won't authenticate when we were actually supposed to

Reply:
Tested this today, and confirmed that password requires a value